Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -86,13 +88,15 @@ class TopSitesPanel: UIViewController {
 
     var profile: Profile! {
         didSet {
-            self.refreshHistory(self.layout.thumbnailCount)
```

Comment:
Should be. I don't really like trusting callers to always do this, but we should take it in the constructor if that's the case. Let me do that (if possible), otherwise I'm happy to do what you're suggesting.